test/distributed/test_c10d.py

```diff
@@ -4218,100 +4218,121 @@ def forward(self, inp):
         process_group = c10d.ProcessGroupNCCL(store, self.rank, self.world_size)
         for use_bucket_view in (True, False):
-            with self.assertRaisesRegex(
-                RuntimeError,
-                "Expected to mark a variable ready only once.",
-            ):
-                model = self._test_ddp_checkpointing(checkpoint_once=False,
-                                                     process_group=process_group,
-                                                     use_bucket_view=use_bucket_view,
-                                                     find_unused_parameters=True)
+            model = self._test_ddp_checkpointing(self.CheckpointTwiceModule(),
+                                                 process_group=process_group,
+                                                 use_bucket_view=use_bucket_view,
+                                                 static_graph=True)
 
     # DDP works as expected if there is weight sharing among layers
     @requires_nccl()
-    @skip_if_not_multigpu
+    @skip_if_lt_x_gpu(2)
     def test_ddp_checkpointing_weight_sharing(self):
         store = c10d.FileStore(self.file_name, self.world_size)
         process_group = c10d.ProcessGroupNCCL(store, self.rank, self.world_size)
         torch.cuda.set_device(self.rank)
-        for use_bucket_view in (True, False):
+        for use_bucket_view, static_graph in product((False, True), (False, True)):
             torch.manual_seed(31415)
-            l1 = nn.Linear(2000, 2000)
-            l2 = nn.Linear(2000, 2000)
+            l1 = nn.Linear(20, 20)
+            l2 = nn.Linear(20, 20)
             l1.weight = l2.weight
-            model = nn.Sequential(l1, l2).cuda()
-            model = nn.parallel.DistributedDataParallel(model,
-                                                        bucket_cap_mb=1,
-                                                        gradient_as_bucket_view=use_bucket_view,
-                                                        device_ids=[self.rank],
-                                                        process_group=process_group)
-            input_tensor = torch.rand((64, 2000), device="cuda", requires_grad=True)
-            output_tensor = checkpoint(model, input_tensor)
-            output_tensor.sum().backward()
-            norm = 0.0
-            for p in model.parameters():
-                self.assertTrue(p.grad is not None)
-                norm += p.grad.norm().item()
-            assert numpy.allclose(norm, 57004), norm
-
+            model = nn.Sequential(l1, l2)
+            self._test_ddp_checkpointing(model,
+                                         process_group=process_group,
+                                         use_bucket_view=use_bucket_view,
+                                         static_graph=static_graph,
+                                         run_checkpoint=True)
+
+    # TODO adding more tests
+    def test_more(self):
```

Comment:
Would be good to have a test where we validate the appropriate errors are raised if the graph doesn't end up being static at some iteration.

Reply:
yeah, sounds good